websocket-tcp-proxy/src/main.rs

```diff
@@ -0,0 +1,112 @@
+use axum::{
+    extract::{
+        ws::{Message, WebSocket, WebSocketUpgrade},
+        Path, TypedHeader,
+    },
+    response::IntoResponse,
+    routing::get,
+    Router,
+};
+use std::net::SocketAddr;
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::TcpStream,
+};
+use tower_http::trace::{DefaultMakeSpan, TraceLayer};
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::fmt::init();
+
+    let app = Router::new().route("/v1/:ip/:port", get(ws_handler)).layer(
+        TraceLayer::new_for_http().make_span_with(DefaultMakeSpan::default().include_headers(true)),
+    );
+
+    let addr = SocketAddr::from(([127, 0, 0, 1], 3001));
```

Comment:
eventually should pick a more random port that is specific to mutiny

Reply:
Made a TODO to let that be configurable